lib/msal-core/tests/mochaTests/genericTests.ts

```diff
@@ -0,0 +1,148 @@
+            expect(url).to.include(DEFAULT_INSTANCE + TENANT + '/oauth2/v2.0/authorize?response_type=id_token&scope=openid%20profile');
+            expect(url).to.include('&client_id=' + MSAL_CLIENT_ID);
+            expect(url).to.include('&redirect_uri=' + encodeURIComponent(msal.getRedirectUri()));
+            expect(url).to.include('&state');
+            expect(url).to.include('&client_info=1');
+            expect(url).not.to.include(Constants.prompt_select_account);
+            expect(url).not.to.include(Constants.prompt_none);
+        });
+        msal.handleRedirectCallbacks(successCallback, errCallback);
+        expect(msal.getRedirectUri()).to.be.equal(TEST_REDIR_URI);
+        msal.loginRedirect({});
+        done();
+    });
+
+    it('navigates user to login and prompt=select_account parameter is passed in request', (done) => {
+        sinon.stub(window.location, "replace").callsFake(function (url) {
+            console.log("New url: " + url);
+            expect(url).to.include(DEFAULT_INSTANCE + TENANT + '/oauth2/v2.0/authorize?response_type=id_token&scope=openid%20profile');
+            expect(url).to.include('&client_id=' + MSAL_CLIENT_ID);
+            expect(url).to.include('&redirect_uri=' + encodeURIComponent(msal.getRedirectUri()));
+            expect(url).to.include('&state');
+            expect(url).to.include('&client_info=1');
+            expect(url).to.include(Constants.prompt_select_account);
+            expect(url).not.to.include(Constants.prompt_none);
+        });
+        msal.handleRedirectCallbacks(successCallback, errCallback);
+        expect(msal.getRedirectUri()).to.be.equal(TEST_REDIR_URI);
+
+        let request: AuthenticationParameters = { prompt: "select_account" };
+        msal.loginRedirect(request);
+        done();
+    });
+
+    it('navigates user to login and prompt=none parameter is passed in request', (done) => {
+        sinon.stub(window.location, "replace").callsFake(function (url) {
+            console.log("New url: " + url);
+            expect(url).to.include(DEFAULT_INSTANCE + TENANT + '/oauth2/v2.0/authorize?response_type=id_token&scope=openid%20profile');
+            expect(url).to.include('&client_id=' + MSAL_CLIENT_ID);
+            expect(url).to.include('&redirect_uri=' + encodeURIComponent(msal.getRedirectUri()));
+            expect(url).to.include('&state');
+            expect(url).to.include('&client_info=1');
+            expect(url).not.to.include(Constants.prompt_select_account);
+            expect(url).to.include(Constants.prompt_none);
+        });
+        msal.handleRedirectCallbacks(successCallback, errCallback);
+        expect(msal.getRedirectUri()).to.be.equal(TEST_REDIR_URI);
+
+        let request: AuthenticationParameters = { prompt: "none" };
+        msal.loginRedirect(request);
+        done();
+    });
+
+    // it('navigates user to redirectURI passed as extraQueryParameter', (done) => {
```

Comment:
lets remove commented out tests

Reply:
This should be resolved in prithvi's latest cut.